redis/lock.test.js

```diff
@@ -0,0 +1,224 @@
+'use strict';
+
+const lockModule = require('./lock'),
+  REDIS_CLIENT = {
+    getAsync: jest.fn().mockResolvedValue(),
+    setAsync: jest.fn().mockResolvedValue(),
+    expire: jest.fn()
+  },
+  fakeGenericErrorLog = jest.fn(),
+  fakeLog = jest.fn(),
+  fakeRedlockInstance = {
+    lock: jest.fn().mockResolvedValue(),
+    unlock: jest.fn().mockResolvedValue()
+  },
+  redlockModule = jest.genMockFromModule('redlock'),
+  bluebirdModule = {
+    delay: jest.fn().mockResolvedValue()
+  };
```

Comment:
Instead of stubbing this bluebird method this way, we can import the whole module in here and just stub the method we want. Here's an example:
```js
const bluebird = require('bluebird'),

// inside a test case
bluebird.delay = jest.fn().mockResolvedValue();
```

Reply:
I didn't know you could do this, thanks! 👍